samples/template_subghz/src/main.c

```diff
@@ -106,13 +112,48 @@ static void app_setup(void)
 	#endif
 }
 
+static app_start_t application_to_start()
+{
+	settings_subsys_init();
+	settings_load();
+
+	#if defined(CONFIG_SIDEWALK_DFU_SERVICE_BLE)
+	bool dfu_mode = false;
+
+	(void) load_immediate_value(CONFIG_DFU_FLAG_SETTINGS_KEY, &DFU_mode, sizeof(DFU_mode));
+
+	if (dfu_mode) {
+		dfu_mode = false;
+		int rc = settings_save_one(CONFIG_DFU_FLAG_SETTINGS_KEY, (const void *)&dfu_mode,
+					   sizeof(dfu_mode));
+		if (rc) {
+			LOG_ERR("Failed to erase DFU flag from persistant storage, Err = %d", rc);
+		}
+		return DFU_APPLICATION;
+	}
+	#endif /* CONFIG_SIDEWALK_DFU_SERVICE_BLE */
+	return SIDEWALK_APPLICATION;
+}
```

Comment:
Q: It looks like duplicated code from samples/template_ble/src/board_events.c, why didn't you exclude it to some common file?

Reply:
Moved to settings_utils